test/integration/framework/shoot_operations.go

```diff
@@ -0,0 +1,270 @@
+	cloudprofileLister := cloudprofileInformer.Lister()
+
+	secretBindingInformer := gardenerTest.K8sGardenInformers.Garden().V1beta1().SecretBindings()
+	secretBindingLister := secretBindingInformer.Lister()
+
+	secretsInformer := gardenerTest.KubeInformerFactory.Core().V1().Secrets()
+	secretsLister := secretsInformer.Lister()
+
+	// Start generating the GardenTestOperation
+	ctx := context.Background()
+	gardenerTest.K8sGardenInformers.Start(ctx.Done())
+
+	if !cache.WaitForCacheSync(ctx.Done(),
+		shootInformer.Informer().HasSynced,
+		seedInformer.Informer().HasSynced,
+		projectInformer.Informer().HasSynced,
+		cloudprofileInformer.Informer().HasSynced,
+		secretBindingInformer.Informer().HasSynced) {
+		panic("Timed out waiting for Garden caches to sync")
+	}
+
+	gardenerTest.KubeInformerFactory.Start(ctx.Done())
+	if !cache.WaitForCacheSync(ctx.Done(), secretsInformer.Informer().HasSynced) {
+		panic("Timed out waiting for Kube caches to sync")
+	}
+
+	return &ShootGardenerTest{
+		GardenerTest: *gardenerTest,
+
+		SecretsInformer: secretsInformer,
+		SecretsLister:   secretsLister,
+
+		SecretBindingInformer: secretBindingInformer,
+		SecretBindingLister:   secretBindingLister,
+
+		CloudProfileInformer: cloudprofileInformer,
+		CloudProfileLister:   cloudprofileLister,
+
+		ProjectInfomer: projectInformer,
+		ProjectLister:  projectLister,
+
+		ShootInformer: shootInformer,
+		ShootLister:   shootLister,
+
+		SeedInformer: seedInformer,
+		SeedLister:   seedLister,
+
+		Shoot:  shoot,
+		Logger: logger,
+	}, nil
+}
+
+// ShootExists returns true if a shoot with the same name exists
+func (s *ShootGardenerTest) ShootExists(ctx context.Context) (bool, *v1beta1.Shoot, error) {
+	//shoot, err := s.K8sGardenClient.Garden().GardenV1beta1().Shoots(s.Shoot.Namespace).Get(s.Shoot.Name, metav1.GetOptions{})
+	shoot, err := s.ShootLister.Shoots(s.Shoot.Namespace).Get(s.Shoot.Name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, shoot, nil
+		}
+		return false, nil, err
+	}
+	return true, shoot, nil
+}
+
+// CreateTestShoot Creates a shoot from a shoot Object
+func (s *ShootGardenerTest) CreateTestShoot(ctx context.Context) (*v1beta1.Shoot, error) {
+	exists, _, err := s.ShootExists(ctx)
+	switch {
+	case err != nil:
+		return nil, err
+	case exists:
+		return nil, fmt.Errorf("shoot already exists")
+	}
+
+	shoot, err := s.K8sGardenClient.Garden().GardenV1beta1().Shoots(s.Shoot.Namespace).Create(s.Shoot)
+	if err != nil {
+		return nil, err
+	}
+
+	// Then we wait for the shoot to be created
+	err = s.WaitForShootToBeCreated(ctx, 30*time.Second, 3600*time.Second)
+	if err != nil {
+		s.Logger.Errorf(err.Error())
+	}
+
+	s.Logger.Infof("Shoot %s was created!", shoot.Name)
+	return shoot, nil
+}
+
+// DeleteTestShoot deletes the test shoot
+func (s *ShootGardenerTest) DeleteTestShoot(ctx context.Context) error {
+	exists, shoot, errShoot := s.ShootExists(ctx)
```

Comment:
This should be just a regular `client.Get`

Reply:
What do you mean by a regular client here?

Reply:
I mean this should be `s.GetShoot` and maybe you wanna have a separate method that, given an `error` can determine the 'error code' of your system (like `ErrShootDoesNotExist`, but maybe the original error anyways is more descriptive as this error constant).